trl/trainer/grpo_trainer.py

```diff
@@ -402,7 +491,7 @@ def __init__(
         args: Optional[GRPOConfig] = None,
         train_dataset: Optional[Union[Dataset, IterableDataset]] = None,
         eval_dataset: Optional[Union[Dataset, IterableDataset, dict[str, Union[Dataset, IterableDataset]]]] = None,
-        processing_class: Optional[PreTrainedTokenizerBase] = None,
+        processing_class: Optional[Union[PreTrainedTokenizerBase, AutoProcessor, ProcessorMixin]] = None,
```

Comment:
In the rest of the trainers with VLM support, instead of `AutoProcessor`, it is `BaseImageProcessor`. I guess that it should be the same here, although I'm not sure about the reasoning tbh.

SFT: 
https://github.com/huggingface/trl/blob/c8347b428798913d7fa4b9516a7b5104f0e1d675/trl/trainer/sft_trainer.py#L356

DPO: 
https://github.com/huggingface/trl/blob/c8347b428798913d7fa4b9516a7b5104f0e1d675/trl/trainer/dpo_trainer.py#L259